lang/test/org/partiql/lang/eval/SimpleEvaluatingCompilerTests.kt

```diff
@@ -125,11 +126,115 @@ class SimpleEvaluatingCompilerTests : EvaluatorTestBase() {
 
     @Test
     fun max() {
+        // Number
         runEvaluatorTestCase("max(`[1, 2, 3]`)", expectedResult = "3")
         runEvaluatorTestCase("max(`[1, 2.0, 3]`)", expectedResult = "3")
         runEvaluatorTestCase("max(`[1, 2e0, 3e0]`)", expectedResult = "3e0")
         runEvaluatorTestCase("max(`[1, 2d0, 3d0]`)", expectedResult = "3d0")
         runEvaluatorTestCase("max(`[1, 2e0, 3d0]`)", expectedResult = "3d0")
         runEvaluatorTestCase("max(`[1, 2d0, 3e0]`)", expectedResult = "3e0")
+
+        // String
+        runEvaluatorTestCase(
+            query = "max(['a', 'abc', '3'])",
+            expectedResult = "'abc'",
+            expectedResultFormat = ExpectedResultFormat.STRING
```

Comment:
For all the tests added here, can we use the [ExpectedResultFormat.ION](https://github.com/partiql/partiql-lang-kotlin/blob/b7e3422d79bfc33ae23ce1a946b9dd6a7156b800/lang/test/org/partiql/lang/eval/evaluatortestframework/ExpectedResultFormat.kt#L10-L16)? It's the strictest (and preferred) option in general for testing the evaluator behavior.

Reply:
I am okay with this suggestion. However, it seems a bit wired to me to write `expectedResult` as `"\"abc\""`. In case that someone does not know the default format is ION instead of STRING, he may think `"abc"` is a double quoted variable.

Reply:
All the tests in this file should use the [ExpectedResultFormat.ION](https://github.com/partiql/partiql-lang-kotlin/blob/b7e3422d79bfc33ae23ce1a946b9dd6a7156b800/lang/test/org/partiql/lang/eval/evaluatortestframework/ExpectedResultFormat.kt#L10-L16). I guess the string escaping could be weird w/ `ExpectedResultFormat.ION`, but we should definitely not use `ExpectedResultFormat.STRING`. From the description of [ExpectedResultFormat.STRING](https://github.com/partiql/partiql-lang-kotlin/blob/b7e3422d79bfc33ae23ce1a946b9dd6a7156b800/lang/test/org/partiql/lang/eval/evaluatortestframework/ExpectedResultFormat.kt#L33-L40):

```
    /**
     * The expected value is an arbitrary string.  [org.partiql.lang.eval.ExprValue.toString]` is called on the result
     * of the query and the expected/actual assertion is performed with regular string equality.
     *
     * This is suboptimal (really, don't use this in new tests) but it is easier to support this here than it is to
     * refactor hundreds of expected values.
     */
    STRING
```

If you're worried about the quote escaping being confusing, you could use the triple-single quotes to not have to escape the strings: ```` ```"foo"```  ````, but I don't think the escaping is confusing.